R/tools-def.R

```diff
@@ -5,87 +5,144 @@ NULL
 #' Define a tool
 #'
 #' @description
-#' Define an R function for use by a chatbot. The function will always be
-#' run in the current R instance.
+#' Annotate a function for use in tool calls, by providing a name, description,
+#' and type defintion for the arguments.
 #'
 #' Learn more in `vignette("tool-calling")`.
 #'
-#' @param .fun The function to be invoked when the tool is called. The return
+#' @param fun The function to be invoked when the tool is called. The return
 #'   value of the function is sent back to the chatbot.
 #'
 #'   Expert users can customize the tool result by returning a
 #'   [ContentToolResult] object.
-#' @param .name The name of the function.
-#' @param .description A detailed description of what the function does.
+#' @param name The name of the function.
+#' @param description A detailed description of what the function does.
 #'   Generally, the more information that you can provide here, the better.
-#' @param .annotations Additional properties that describe the tool and its
+#' @param annotations Additional properties that describe the tool and its
 #'   behavior. Usually created by [tool_annotations()], where you can find a
 #'   description of the annotation properties recommended by the [Model Context
 #'   Protocol](https://modelcontextprotocol.io/introduction).
-#' @param .convert Should JSON inputs be automatically convert to their
+#' @param convert Should JSON inputs be automatically convert to their
 #'   R data type equivalents? Defaults to `TRUE`.
-#' @param ... Name-type pairs that define the arguments accepted by the
+#' @param arguments A named list that define the arguments accepted by the
 #'   function. Each element should be created by a [`type_*()`][type_boolean]
 #'   function.
+#' @param ... `r lifecycle::badge("deprecated")` Use `arguments` instead.
+#' @param .name,.description,.convert,.annotations
+#'   `r lifecycle::badge("deprecated")` Please switch to the non-prefixed
+#'   equivalents.
 #' @return An S7 `ToolDef` object.
 #' @examplesIf has_credentials("openai")
 #'
 #' # First define the metadata that the model uses to figure out when to
 #' # call the tool
 #' tool_rnorm <- tool(
 #'   rnorm,
-#'   "Drawn numbers from a random normal distribution",
-#'   n = type_integer("The number of observations. Must be a positive integer."),
-#'   mean = type_number("The mean value of the distribution."),
-#'   sd = type_number("The standard deviation of the distribution. Must be a non-negative number."),
-#'   .annotations = tool_annotations(
```

Comment:
I dropped `annotations` here since they're not currently used, and I don't want folks to think they need to define just to use a tool.